shadowsocks-csharp/Model/NlogConfig.cs

```diff
@@ -108,11 +108,20 @@ private static XmlNode SelectSingleNode(XmlDocument doc, string xpath)
         /// </summary>
         public static void TouchAndApplyNLogConfig()
         {
-            if (!File.Exists(NLOG_CONFIG_FILE_NAME))
+            try
             {
+                if (File.Exists(NLOG_CONFIG_FILE_NAME))
+                    return; // NLog.config exists, and has already been loaded
+
                 File.WriteAllText(NLOG_CONFIG_FILE_NAME, Properties.Resources.NLog_config);
-                LogManager.LoadConfiguration(NLOG_CONFIG_FILE_NAME);
             }
+            catch (Exception ex)
+            {
+                NLog.Common.InternalLogger.Error(ex, "[shadowsocks] Failed to setup default NLog.config: {0}", NLOG_CONFIG_FILE_NAME);
+                return;
+            }
+
+            LoadConfiguration();    // Load the new config-file

```

Comment:
There may not have the NLog config file due to the previous write permission issue.

Reply:
It only performs the `LoadConfiguration`  if it was successful in creating one. So if it can write to the location, then it should also be able to read.

Reply:
So the routine is like this? I think this change is ok then.

When write error: Line 113->116 throw error ->118 catch error->120->121 return
When config exist: (config auto loaded by NLog itself at program start) 113->114 return
When config not exist: 113->116->124 load config->return

Reply:
Yes it returns without doing anything for all cases. Except for the very first time where it successfully have created a new default `NLog.config`.

Reply:
Since the 5bc4a0b1cc595b83f81149bf656de48f93758296 is committed to fix the CurrentDirectory issue. The NLog config file is always create/read/write on the same place in file system. Will merge the code soon.

PS:
Before the previous autostart issue fix, SS may create the NLog config file in the wrong place, such as `System` folder and request the NLog to load it.
Then the following code change the CurrentDirectory to `Application.StartPath`
https://github.com/shadowsocks/shadowsocks-windows/blob/5bc4a0b1cc595b83f81149bf656de48f93758296/shadowsocks-csharp/Program.cs#L86
Which made the NLog file location inconsistent. When user change the log level and NLog need to reload the config file, may fail.